[ACM-12699]: Updating HCP multi-arch docs (#6692)
open-cluster-management/rhacm-docs: clusters/hosted_control_planes/enable_hcp_on_arm.adoc

Reviewer: On L4, it says `You can enable an ARM64 hosted control plane to operate with an {ocp-short} ARM64 data plane in a management cluster.`

I don't know if we have another spot in docs saying this, but you can have ARM64 NodePools from a AMD64 hosted control plane too.

Reviewer: Also on L4, maybe we should say you can create either an `amd64` or `arm64` nodepool from a Arm64 HCP?

[#hosted-cluster-arm-aws]
= Enabling hosted clusters to run on different architectures (Technology Preview)

By default for hosted control planes on AWS, you use an AMD64 hosted cluster. However, you can enable hosted control planes to run on an ARM64 hosted cluster. 

> You can also create node pools of different architecture types for the same hosted cluster. The following table shows compatible combinations of node pools and hosted clusters.

Reviewer 2: Optional, you could be a bit more user focused and direct the user to view the table:

```suggestion
You can also create node pools of different architecture types for the same hosted cluster. See the following table for compatible combinations of node pools and hosted clusters:
```

Author: Good suggestion! Will implement.

[#hosted-cluster-arm-aws]
= Enabling hosted control planes on an ARM64 {ocp-short} cluster (Technology Preview)

> You can enable an ARM64 or AMD64 hosted control plane to operate with an {ocp-short} ARM64 data plane in a management cluster. You can also create an ARM64 or AMD64 node pool from an ARM64 hosted control plane.

Reviewer: Should the last `ARM64` in this sentence actually be `AMD64`? The title on L2 references ARM64 mgmt cluster.

Author: Good catch. Do you think we should update the title on L2 to include AMD64 as well? Maybe something like this? `Enabling hosted control planes on an ARM64 or AMD64 OpenShift Container Platform cluster`

Reviewer: That's fine with me but are both Tech Preview or just one combination? That's where it might get confusing on what support level is applicable for which combo.

Author: Nevermind. Adel cleared that up for me. The title will stay as is.

> You can enable an ARM64 or AMD64 hosted control plane to operate with an {ocp-short} ARM64 data plane in a management cluster. You can also create an ARM64 or AMD64 node pool from an AMD64 hosted control plane.

Reviewer: ```suggestion
You can enable an ARM64 hosted control plane to operate with an {ocp-short} ARM64 data plane in a management cluster. You can create ARM64 or AMD64 node pools from this ARM64 hosted control plane.
```

Reviewer: > I don't know if we have another spot in docs saying this, but you can have ARM64 NodePools from a AMD64 hosted control plane too.

What I was trying to say here previously, was this PR is just for "Enabling hosted control planes on an ARM64 {ocp-short} cluster (Technology Preview)". It is also true that you can have amd64 or arm64 nodepools from a AMD64 cluster as well. I don't know if that is called out somewhere; it's not only valid on a ARM64 cluster, if that makes sense.

--pull-secret <path_to_pull_secret> \ <4>
--sts-creds <path_to_sts_credential_file> \ <5>
--region <region> \ <6>
--release-image quay.io/openshift-release-dev/ocp-release:<ocp_release_image> \ <7>
--role-arn <role_name> \ <8>

> --multi-arch <9>

Reviewer: ```suggestion
--multi-arch <9>
--arch <10>
```

where <10> is arm64 (but can also be amd64)

<6> Specify the AWS region name, for example, `us-east-1`.
<7> Specify the supported {ocp-short} version that you want to use, for example, `4.14.0-x86_64`. If you are using a disconnected environment, replace `<ocp_release_image>` with the digest image. To extract the {ocp-short} release image digest, see _Extracting the {ocp-short} release image digest_.
<8> Specify the Amazon Resource Name (ARN), for example, `arn:aws:iam::820196288204:role/myrole`.

> <9> The `multi-arch` flag indicates that the hosted cluster supports both `amd64` and `arm64` node pools. When this flag is set and a release image is supplied, the release image must be a multi-architecture release image. When this flag is set and a release stream is supplied, the release stream must be a multi-architecture release stream. If you use a multi-architecture release image or stream and you do not set the `multi-arch` flag, the management cluster and node pool CPU arches must match. Otherwise, the CLI throws a validation error and does not create a hosted cluster. A node pool supports only one CPU architecture within the same node pool.

Author: Not sure about the term _arches_ here. I based this description on the upstream docs. Should _arches_ be _architectures_?

Reviewer: Yeah it can be. arches == architectures.

Reviewer: I need to update the upstream docs. We decided not to throw a validation error but warn the user if the multi-arch flag isn't set and the mgmt cluster worker nodes CPU architecture do not match the nodepools CPU architecture.

> . Add a 64-bit x_86 `NodePool` object to the hosted cluster by running the following command:

Reviewer: Do we want a separate section for how to add a amd64 nodepool and a section for how to add an arm64 nodepool or could we just combine the two together? Basically line 100 down.

Author: I tried to combine them. Take a look when you can and let me know what you think.

Reviewer: Looks good to me.

<6> Specify the AWS region name, for example, `us-east-1`.
<7> Specify the supported {ocp-short} version that you want to use, for example, `4.14.0-x86_64`. If you are using a disconnected environment, replace `<ocp_release_image>` with the digest image. To extract the {ocp-short} release image digest, see _Extracting the {ocp-short} release image digest_.
<8> Specify the Amazon Resource Name (ARN), for example, `arn:aws:iam::820196288204:role/myrole`.

> <9> The `multi-arch` flag indicates that the hosted cluster supports both `amd64` and `arm64` node pools. When this flag is set and a release image is supplied, the release image must be a multi-architecture release image. When this flag is set and a release stream is supplied, the release stream must be a multi-architecture release stream. If you use a multi-architecture release image or stream and you do not set the `multi-arch` flag, the management cluster and node pool CPU architectures must match. Otherwise, a warning is displayed. A node pool supports only one CPU architecture within the same node pool.

Reviewer: ```suggestion
<9> The `multi-arch` flag indicates that the hosted cluster supports both `amd64` and `arm64` node pools. When this flag is set and a release image is supplied, the release image must be a multi-architecture release image. When this flag is set and a release stream is supplied, the release stream must be a multi-architecture release stream. If you do not set the `multi-arch` flag, the management cluster and node pool CPU architectures must match. Otherwise, a warning is displayed. A node pool supports only one CPU architecture within the same node pool.
```

Author: Done

<6> Specify the AWS region name, for example, `us-east-1`.
<7> Specify the supported {ocp-short} version that you want to use, for example, `4.14.0-x86_64`. If you are using a disconnected environment, replace `<ocp_release_image>` with the digest image. To extract the {ocp-short} release image digest, see _Extracting the {ocp-short} release image digest_.
<8> Specify the Amazon Resource Name (ARN), for example, `arn:aws:iam::820196288204:role/myrole`.

> <9> The `multi-arch` flag indicates that the hosted cluster supports both `amd64` and `arm64` node pools. When this flag is set and a release image is supplied, the release image must be a multi-architecture release image. When this flag is set and a release stream is supplied, the release stream must be a multi-architecture release stream. If you do not set the `multi-arch` flag, the management cluster and node pool CPU architectures must match. Otherwise, a warning is displayed. A node pool supports only one CPU architecture within the same node pool.

Reviewer 2: Optional, if the user is the one "setting" you could write that here: 

```suggestion
<9> The `multi-arch` flag indicates that the hosted cluster supports both `amd64` and `arm64` node pools. When you set the flag and supply a release image, the release image must be a multi-architecture release image. When you set the flag and supply a release stream, the release stream must be a multi-architecture release stream. If you do not set the `multi-arch` flag, the management cluster and node pool CPU architectures must match. Otherwise, a warning is displayed. A node pool supports only one CPU architecture within the same node pool.
```

Author: Nice! I'll update the wording according to your suggestion.

If you do not specify a value for the `arch` field, the `amd64` value is used by default.

[#hc-arm-nodepools-prereqs]
=== Prerequisites

You must have a multi-architecture image for the `HostedCluster` custom resource to use. You can access multi-architecture nightly images at https://multi.ocp.releases.ci.openshift.org/.

[#hc-arm-nodepools-aws]
=== Creating an ARM or AMD NodePool object on an AWS hosted cluster

To create an ARM or AMD `NodePool` object on a hosted cluster on AWS, complete the following steps:

> . Render the `NodePool` object configuration in a YAML format by running the following command:

Reviewer: I think you can delete everything from L82 to L101.

Author: Removed

<3> Specify the node pool replica count, for example, `3`.
<4> The `NodePool` object configuration is saved in the `arm64-nodepool.yml` file.

> . Add a 64-bit x_86 `NodePool` object to the hosted cluster by running the following command:

Reviewer: ```suggestion
. Add a `NodePool` object to the hosted cluster by running the following command:
```
